pkgs/tools/misc/btdu/default.nix

```diff
@@ -0,0 +1,76 @@
+{stdenv, lib, fetchurl, dub, ncurses, ldc, zlib, removeReferencesTo }:
+
+let
+    _d_ae_ver              = "0.0.3100";
+    _d_btrfs_ver           = "0.0.12";
+    _d_ncurses_ver         = "0.0.149";
+    _d_emsi_containers_ver = "0.9.0";
+in
+stdenv.mkDerivation rec {
+    pname = "btdu";
+    version = "0.3.1";
+
+    srcs = [
+      (fetchurl {
+        url = "https://github.com/CyberShadow/${pname}/archive/v${version}.tar.gz";
+        sha256 = "760b2f0d28920a78b7967dd34c429125135688a3aefc57ab3a92d07bc3ef10cb";
+      })
+      (fetchurl {
+        url = "https://github.com/CyberShadow/ae/archive/v${_d_ae_ver}.tar.gz";
+        sha256 = "86fa09ef6c1be4cbe8ad1c85729054e5d691b41ff57c7980d99937ec0f45b950";
+      })
+      (fetchurl {
+        url = "https://github.com/CyberShadow/d-btrfs/archive/v${_d_btrfs_ver}.tar.gz";
+        sha256 = "cf2b1fa3e94a0aa239d465adbac239514838835283521d632f571948aa517f92";
+      })
+      (fetchurl {
+        url = "https://github.com/D-Programming-Deimos/ncurses/archive/v${_d_ncurses_ver}.tar.gz";
+        sha256 = "2c8497f5dd93f9d3a05ca7ed57c4fcaee1e988fd25a24de106917ddf72f34646";
+      })
+      (fetchurl {
+        url = "https://github.com/dlang-community/containers/archive/v${_d_emsi_containers_ver}.tar.gz";
+        sha256 = "5e256b84bbdbd2bd625cba0472ea27a1fde6d673d37a85fe971a20d52874acaa";
+      })
+    ];
+
+    sourceRoot = ".";
+
+    postUnpack = ''
+      mv ae-${_d_ae_ver} "ae"
+    '';
+
+
+    nativeBuildInputs = [ dub ldc ];
+    buildInputs = [ ncurses zlib ];
+
+    configurePhase = ''
+      mkdir home
+      HOME="home" dub add-local ae ${_d_ae_ver}
+      HOME="home" dub add-local d-btrfs-${_d_btrfs_ver} ${_d_btrfs_ver}
+      HOME="home" dub add-local ncurses-${_d_ncurses_ver} ${_d_ncurses_ver}
+      HOME="home" dub add-local containers-${_d_emsi_containers_ver} ${_d_emsi_containers_ver}
+    '';
+
+    buildPhase = ''
+      cd ${pname}-${version}
+      HOME="../home" dub --skip-registry=all build -b release
+    '';
+
+    installPhase = ''
+      mkdir -p $out/bin
+      cp btdu $out/bin/
+    '';
+
+    postInstall = ''
+      ${removeReferencesTo}/bin/remove-references-to -t ${stdenv.cc} $(readlink -f $out/lib/libdruntime-ldc-shared)
```

Comment:
```suggestion
      ${removeReferencesTo}/bin/remove-references-to -t ${ldc} $(readlink -f $out/lib/libdruntime-ldc-shared)
```
(to be tested)

Reply:
I will test right away!

Reply:
I've tested here and it still references ldc:
```
/nix/store/jcb7fny2k03pfbdqk1hcnh12bxgax6vf-glibc-2.33-108
/nix/store/3y4g885qcv7wmzczcgiz4xgshy75snd3-ncurses-6.3
/nix/store/92wmg8niqrvwxsfigh5hhjpb0y9ska1i-zlib-1.2.11
/nix/store/dphabgk0ha7pzp4ca9sdry6fhvpin1lv-curl-7.81.0
/nix/store/z85pkakmhsj4pv8n44y7pw55hqvqn4zy-tzdata-2021e
/nix/store/4i6ff0409wip24r1hcjmych3mylw6i9s-ldc-1.27.1
```
this $out/lib/libdruntime-ldc-shared that actually confuses me the most.

Reply:
the second argument is the file that contains the reference. Either there is only one such file (find it with `nix why-depends`) or you can make a loop with find: random example found with grep:
`find $out -type f -exec ${removeReferencesTo}/bin/remove-references-to -t ${go} '{}' +`